### PR TITLE
add versioned finality proof decoding

### PR DESCRIPTION
**Pull request by oshorefueled**
This PR: 
* adds version finality proof decoding and encoding.
* adds testing for decoding version finality proof
* It also corrects the deserialization of justifications in SignedBlock.

### CODE REVIEW

File: types/extrinsic.go [hunk 1]
```diff
@@ -334,10 +334,7 @@ func (a *Args) Decode(decoder scale.Decoder) error {
 	return nil
 }
 
-type Justification struct {
-	ConsensusEngineID    ConsensusEngineID
-	EncodedJustification Bytes
-}
+type Justification [2][]byte
```
Comment (seunlanlege) [inline types/extrinsic.go:337]: This isn't the correct structure of a `Justification`

Why did you change it?

File: types/versioned_finality_proof.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+package types
+
+import (
+	"fmt"
+
+	"github.com/ComposableFi/go-substrate-rpc-client/v4/scale"
+)
+
+// DigestItem specifies the item in the logs of a digest
```
Comment (seunlanlege) [inline types/versioned_finality_proof.go:25]: Incorrect docs

File: types/versioned_finality_proof_test.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+		147, 45, 233, 2, 241, 172, 29, 48, 242, 54, 128, 203, 198, 120, 143, 117, 51, 77, 227, 31, 54, 66, 98, 178, 35, 248, 131, 143, 225, 186, 79, 181,
+		41, 139, 122, 101, 192, 52, 71, 146, 110, 143, 241, 255, 39, 12, 154, 113, 1, 187, 0, 141, 146, 245, 139, 117, 115, 190, 149, 122, 94, 155, 213,
+		208, 33, 38, 41, 195, 25, 231, 187, 251, 41, 246, 250, 207, 251, 29, 38, 128, 156, 27, 9, 223, 106, 226, 84, 74, 178, 133, 227, 227, 86, 12, 84,
+		49, 25, 153, 43, 124, 240, 170, 196, 232, 252, 113, 9, 20, 54, 224, 38, 63, 14, 0, 218, 30, 250, 99, 133, 109, 91, 223, 88, 105, 130, 178, 176,
+		197, 131, 237, 98, 201, 69, 129, 71, 174, 46, 64, 130, 113, 245, 238, 246, 144, 145, 79, 53, 169, 73, 192, 95, 101, 203, 140, 70, 144, 56, 54,
+		83, 170, 203, 86, 181, 182, 135, 146, 192, 71, 74, 185, 61, 219, 171, 53, 71, 146, 240, 40, 0, 251, 252, 51, 163, 152, 68, 134, 160, 70, 183,
+		88, 12, 25, 167, 121, 9, 199, 201, 15, 194, 163, 195, 170, 142, 206, 155, 147, 67, 13, 8, 242, 125, 2, 96, 232, 10, 111, 111, 241, 221, 97, 101,
+		234, 133, 53, 250, 160, 14, 190, 23, 67, 76, 29, 80, 133, 130, 71, 17, 242, 176, 163, 37, 4, 187, 0,
```
Comment (seunlanlege) [inline types/versioned_finality_proof_test.go:19]: Add a test for decoding the `CompactSignedCommitment` from the encoded justification

File: types/extrinsic.go (hunk 1 above)
Comment (oshorefueled) [reply types/extrinsic.go:337]: I got an error about decoding justifications into the previous type. `json: cannot unmarshal array into Go struct field SignedBlock.justifications of type types.Justification`.

Looking at the structure of the Commitment object, I saw that justifications was a multidimensional array.
```json
  justifications: [
    [
      FRNK
      0x1200000000000000b356faf52de19d181c11f094bb322a0820490616366e64d6caa6a1dedf4fd7b81500000010b356faf52de19d181c11f094bb322a0820490616366e64d6caa6a1dedf4fd7b815000000c4d793c8a2d29de10baff4b6e5222dfd9e104161ed8d3226bc514407f1e17c8461325e385f006d39b53c65d937701f492ff54f8efbabf725ce3b1d08e2bf6d0b568cb4a574c6d178feb39c27dfc8b3f789e5f5423e19c71633c748b9acf086b5b356faf52de19d181c11f094bb322a0820490616366e64d6caa6a1dedf4fd7b8150000007f05e9fcef3f2c61aad0217fb9ab1fe1a8b93910a0f3a7300f3c2eebd4a2af6d565beeeec95b816679393ed12fec2b81df560ebdf8d3e26597079e0cf002b2015e639b43e0052c47447dac87d6fd2b6ec50bdd4d0f614e4299c665249bbd09d9b356faf52de19d181c11f094bb322a0820490616366e64d6caa6a1dedf4fd7b8150000007efdbb3bd4627187959c20a087fb949be9433eda19b1ac3b47170593be9471d56230fb26c5fb028aa870b4d94e8a36382bac100739524d22211129d999b3a50388dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0eeb356faf52de19d181c11f094bb322a0820490616366e64d6caa6a1dedf4fd7b8150000005f4696b936c7e7d64e1a0b90600ddc5d57a2cdd3bb149557e778b0aeb6e151665d2330bf13a4f6ed7f9af40a052505dd49afd24ac077f93a5a2d83d51fd8c60dd17c2d7823ebf260fd138f2d7e27d114c0145d968b5ff5006125f2414fadae6900
    ]
    [
      BEEF
      0x01046d6880f16dc85b041fcc08f578de648909bb24468bbecb6a35f86932cb9ba1b47795bd15000000020000000000000004f00500000010d6b8752891ad3027e252083e007e31fe14a40589c6166a227880e9ec3d59291d62868293cb01303edecc116c2774a158d23b59f10f1f2017b269a6e682d33a5101ebe6158eb21eed20e644b7c46a4046950ab86def9d609d16797e5fcbf138017c77c0c0c28e7b86a43e9d498bc28f842a8dc7c593e12c4601427ee96de5e48d2101e593acc24a110f90a97f16bdba802244fb4b47cac7155f6b788abd52a18d897d71ccef729d19e21b8c31687bb1f1198ebcd54c301ac8c30f7d5c943425243f6d011d3107c91f184a0c4be8b2a03e2f7ee3be1b185436de3f5e2c797aebf0581bf6253e34e6e42e44fe0b76ebf681111e35727c60a44a1e526059591a97b9b57ff801
    ]
  ]
```

Comment (seunlanlege) [reply types/extrinsic.go:337]: Json? That's not right.

File: types/versioned_finality_proof_test.go (hunk 3 above)
Comment (oshorefueled) [reply types/versioned_finality_proof_test.go:19]: Do you mean calling the unpack method? Because the CompactSignedCommitment is already decode into the AsCompactSignedCommitment field in the VersionedFinalityProof type.

File: types/extrinsic.go [hunk 4]
```diff
@@ -339,7 +339,8 @@ type Justification struct {
 	EncodedJustification Bytes
 }
 
-type Justifications []Justification
+// TODO: unmarshal this [2][]byte into Justification
```
Comment (seunlanlege) [inline types/extrinsic.go:342]: @oshorefueled Left this for you. Once this is sorted we're good to merge
